### PR TITLE
Gym opacity

### PR DESCRIPTION
**Pull request by springjools**
#Fade out gyms that have been scanned a long time ago

## Description
Gyms scanned less than 2 hours ago: 100% opacity
Gyms scanned more than 2 hours ago but less than 12 hours ago: 75% opacity
Gyms scanned more than 12 hours ago but less than 24 hours ago: 33% opacity
Older gyms: 17 % opacity

Can be configured in  map.common.js
## Motivation and Context
Makes it easier to distinguish old gym data from new one

## How Has This Been Tested?
Tested on local machine

## Screenshots (if appropriate):
Normal map: http://prntscr.com/dup4jx
Dark map: http://prntscr.com/dup6nu

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by springjools** [comment]
Yup, the opacity levels and times are of course up for discussion. I just tested quickly and chose these values.

**Comment by joekeating87** [comment]
Just an idea, but can you make the times, have default, but be customized in map.common.js?

**Comment by springjools** [comment]
@averylarry : It's not hard to make more levels, but I tested with different opacities and my experience was that it's not very easy to quickly see the difference between them, unless they differ by a large amount. That's why I chose 15 %, 40 % and 100%. But it could be just my screen/eyes, others are welcome to give their opinions...

**Comment by drallieiv** [comment]
need to fix indentation and spaces

File: static/js/map.js [hunk 1]
```diff
@@ -718,19 +719,22 @@ function setupGymMarker(item) {
 }
 
 function updateGymMarker(item, marker) {
-    marker.setIcon({
-        url: 'static/forts/' + Store.get('gymMarkerStyle') + '/' + gymTypes[item['team_id']] + (item['team_id'] !== 0 ? '_' + getGymLevel(item['gym_points']) : '') + '.png',
-        scaledSize: new google.maps.Size(48, 48)
-    })
+    var timeDelta = (Date.now() - item['last_scanned']) / 1000 / 60 // minutes since last scan
```
Comment (onilton) [inline static/js/map.js]: This code is repeated (from above), maybe isolate it in function `getOpacity(item)` or `getGymOpacity(item)`?

File: static/js/map.js [hunk 2]
```diff
@@ -662,16 +662,17 @@ function customizePokemonMarker(marker, item, skipNotification) {
 }
 
 function setupGymMarker(item) {
+    var timeDelta = (Date.now() - item['last_scanned']) / 1000 / 60 // minutes since last scan
+    var gymSize = Store.get('gymMarkerSize')
+    var opacity = (timeDelta < Store.get('obsoletion1')) ? 1.0 : (timeDelta < Store.get('obsoletion2')) ? Store.get('opacity1') : (timeDelta < Store.get('obsoletion3')) ? Store.get('opacity2') : Store.get('opacity3')
```
Comment (onilton) [inline static/js/map.js]: This code is repeated (from above), maybe isolate it in function `getOpacity(item)` or `getGymOpacity(item)`?

File: static/js/map.js [hunk 3]
```diff
@@ -718,19 +719,22 @@ function setupGymMarker(item) {
...
+    var gymSize = Store.get('gymMarkerSize')
+
+    marker.setOpacity(opacity)
+    marker.setIcon({url: 'static/forts/' + Store.get('gymMarkerStyle') + '/' + gymTypes[item['team_id']] + (item['team_id'] !== 0 ? '_' + getGymLevel(item['gym_points']) : '') + '.png', scaledSize: new google.maps.Size(gymSize, gymSize)})
     marker.infoWindow.setContent(gymLabel(gymTypes[item['team_id']], item['team_id'], item['gym_points'], item['latitude'], item['longitude'], item['last_scanned'], item['name'], item['pokemon'], item['gym_id']))
     return marker
 }
 
 function updateGymIcons() {
+    var gymSize = Store.get('gymMarkerSize')
```
Comment (onilton) [inline static/js/map.js]: Is gymSize really required for this PR? I would remove it if not really necessary for gym opacity. (Commented here, but there are other parts of the code that use it)

Comment (Arengo) [inline static/js/map.js]: Defending the ``gymMarkerSize``, it was hard coded in map.js instead having a default value in ``map.common.js``. I like these changes, making easier customizable in ``map.common.js``

**Comment by SiteEffect** [comment]
Would also like to test it for an approval as soon as @onilton's review is handled.

**Comment by springjools** [comment]
I don't disagree with any changes but I will handle this and other pr:s together in order to avoid micromanaging individual ones, especially waiting to clarify status on #1722

**Comment by sebastienvercammen** [comment]
With recent changes to raids and a decreased interest in the PR, this PR will be closed until enough people want this merged.

**Pull request closed without merging**